### ISSUE TITLE
Arrow Flight SQL server AuthN and AuthZ

### ISSUE THREAD

**Issue by Susmit07** [apache/arrow#44456]
### Describe the usage question you have. Please include as many useful details as  possible.

Hi Team,

If things goes well, we will be using Arrow Flight SQL Server. 

We will be publishing sdks for java and python for teams to consume in our organisation.

Flight SQL server is written in Python and we are using DuckDB query engine for query support on S3 parquet files.

```
sql_query = "SELECT * FROM read_parquet('s3a://bucket/parquets/flights-1m-new-*.parquet');"
```

The performance numbers are really good.

I want to know how to authenticate and authorize a user (best practices) accessing a parquet data for a bucket. Every teams will have dedicated S3 buckets.

Everytime initialising s3 client with new access and secret keys won't it be resource intensive?

### Component(s)

FlightRPC

**Comment by lidavidm** [comment]
Isn't this question more about DuckDB? Flight isn't involved in the S3 access here.

**Comment by lidavidm** [comment]
If you just want authentication/authorization, you can implement a middleware to check a bearer token or similar.

**Comment by Susmit07** [comment]
Yeah authN i am thinking to use a JWT based authN, you are correct on accessing the s3 buckets by duck db can we initialise s3 client with new access and secret keys every-time do you see any issues here?

**Comment by Susmit07** [comment]
sure will do.. thank @lidavidm
